pandas/tests/indexes/datetimes/test_timezones.py

```diff
@@ -0,0 +1,1047 @@
+                                    gettz('US/Eastern')])
+    def test_dti_tz_localize_ambiguous_nat(self, tz):
+        times = ['11/06/2011 00:00', '11/06/2011 01:00', '11/06/2011 01:00',
+                 '11/06/2011 02:00', '11/06/2011 03:00']
+        di = DatetimeIndex(times)
+        localized = di.tz_localize(tz, ambiguous='NaT')
+
+        times = ['11/06/2011 00:00', np.NaN, np.NaN, '11/06/2011 02:00',
+                 '11/06/2011 03:00']
+        di_test = DatetimeIndex(times, tz='US/Eastern')
+
+        # left dtype is datetime64[ns, US/Eastern]
+        # right is datetime64[ns, tzfile('/usr/share/zoneinfo/US/Eastern')]
+        tm.assert_numpy_array_equal(di_test.values, localized.values)
+
+    @pytest.mark.parametrize('tz', [pytz.timezone('US/Eastern'),
+                                    gettz('US/Eastern')])
+    def test_dti_tz_localize_ambiguous_flags(self, tz):
+        # November 6, 2011, fall back, repeat 2 AM hour
+
+        # Pass in flags to determine right dst transition
+        dr = date_range(datetime(2011, 11, 6, 0), periods=5,
+                        freq=pd.offsets.Hour(), tz=tz)
+        times = ['11/06/2011 00:00', '11/06/2011 01:00', '11/06/2011 01:00',
+                 '11/06/2011 02:00', '11/06/2011 03:00']
+
+        # Test tz_localize
+        di = DatetimeIndex(times)
+        is_dst = [1, 1, 0, 0, 0]
+        localized = di.tz_localize(tz, ambiguous=is_dst)
+        tm.assert_index_equal(dr, localized)
+        tm.assert_index_equal(dr, DatetimeIndex(times, tz=tz,
+                                                ambiguous=is_dst))
+
+        localized = di.tz_localize(tz, ambiguous=np.array(is_dst))
+        tm.assert_index_equal(dr, localized)
+
+        localized = di.tz_localize(tz,
+                                   ambiguous=np.array(is_dst).astype('bool'))
+        tm.assert_index_equal(dr, localized)
+
+        # Test constructor
+        localized = DatetimeIndex(times, tz=tz, ambiguous=is_dst)
+        tm.assert_index_equal(dr, localized)
+
+        # Test duplicate times where infer_dst fails
+        times += times
+        di = DatetimeIndex(times)
+
+        # When the sizes are incompatible, make sure error is raised
+        with pytest.raises(Exception):
+            di.tz_localize(tz, ambiguous=is_dst)
+
+        # When sizes are compatible and there are repeats ('infer' won't work)
+        is_dst = np.hstack((is_dst, is_dst))
+        localized = di.tz_localize(tz, ambiguous=is_dst)
+        dr = dr.append(dr)
+        tm.assert_index_equal(dr, localized)
+
+        # When there is no dst transition, nothing special happens
+        dr = date_range(datetime(2011, 6, 1, 0), periods=10,
+                        freq=pd.offsets.Hour())
+        is_dst = np.array([1] * 10)
+        localized = dr.tz_localize(tz)
+        localized_is_dst = dr.tz_localize(tz, ambiguous=is_dst)
+        tm.assert_index_equal(localized, localized_is_dst)
+
+    # TODO: belongs outside tz_localize tests?
```

Comment:
yeah move to construction

Reply:
See the "FIXME" note about 4 lines down.  I'm not confident this test is testing what it is intending. This is where the existing indentation typo is: https://github.com/pandas-dev/pandas/pull/19545/files#diff-f67205b3c64b63fa7471c4221e413a19L424